### ISSUE TITLE
Rollup lost an else statement (tree shaking away a conditional function inside a loop)

### ISSUE THREAD

**Issue by bhousel** [rollup/rollup#2302]
Not sure what is causing this `else` block to go away.  Rollup seems to tree shake away the `isInteresting` function in some situations.

[View the repro in the repl](https://rollupjs.org/repl?version=0.62.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMGlzSW50ZXJlc3RpbmdUYWclMjAlN0QlMjBmcm9tJTIwJy4lMkZpbnRlcmVzdGluZy5qcyclM0IlNUNuJTVDbnZhciUyMHRvQ2hlY2slMjAlM0QlMjAlNUIlN0JoaWdod2F5JTNBJTIwJ3ByaW1hcnknJTdEJTJDJTIwJTdCaGlnaHdheSUzQSUyMCdzZWNvbmRhcnknJTdEJTVEJTNCJTVDbnZhciUyMHRhZ3MlMjAlM0QlMjAlN0IlN0QlM0IlNUNudmFyJTIwY29uZmxpY3RpbmclMjAlM0QlMjBmYWxzZSUzQiU1Q24lNUNudG9DaGVjay5mb3JFYWNoKGZ1bmN0aW9uKHNyYyklMjAlN0IlNUNuJTVDdGZvciUyMCh2YXIlMjBrJTIwaW4lMjBzcmMpJTIwJTdCJTVDbiU1Q3QlNUN0aWYlMjAoIShrJTIwaW4lMjB0YWdzKSklMjAlN0IlNUNuJTVDdCU1Q3QlNUN0JTVDdHRhZ3MlNUJrJTVEJTIwJTNEJTIwc3JjJTVCayU1RCUzQiU1Q24lNUN0JTVDdCU3RCUyMGVsc2UlMjBpZiUyMCh0YWdzJTVCayU1RCUyMCUyNiUyNiUyMGlzSW50ZXJlc3RpbmdUYWcoayklMjAlMjYlMjYlMjB0YWdzJTVCayU1RCUyMCElM0QlM0QlMjBzcmMlNUJrJTVEKSUyMCU3QiU1Q24lNUN0JTVDdCU1Q3QlNUN0Y29uZmxpY3RpbmclMjAlM0QlMjB0cnVlJTNCJTVDbiU1Q3QlNUN0JTdEJTVDbiU1Q3QlN0QlNUNuJTdEKSUzQiU1Q24lNUNuY29uc29sZS5sb2coY29uZmxpY3RpbmcpJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmludGVyZXN0aW5nLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwaXNJbnRlcmVzdGluZ1RhZyhrZXkpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMGtleSUyMCElM0QlM0QlMjAnYXR0cmlidXRpb24nJTIwJTI2JTI2JTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGtleSUyMCElM0QlM0QlMjAnY3JlYXRlZF9ieSclMjAlMjYlMjYlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwa2V5JTIwISUzRCUzRCUyMCdzb3VyY2UnJTIwJTI2JTI2JTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGtleSUyMCElM0QlM0QlMjAnb2RibCclMjAlMjYlMjYlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwa2V5LmluZGV4T2YoJ3RpZ2VyJTNBJyklMjAhJTNEJTNEJTIwMCUzQiU1Q24lNUNuJTdEJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

<details>

<summary>summary</summary>

### main.js
```js
import { isInterestingTag } from './interesting.js';

var toCheck = [{highway: 'primary'}, {highway: 'secondary'}];
var tags = {};
var conflicting = false;

toCheck.forEach(function(src) {
	for (var k in src) {
		if (!(k in tags)) {
				tags[k] = src[k];
		} else if (tags[k] && isInterestingTag(k) && tags[k] !== src[k]) {
				conflicting = true;
		}
	}
});

console.log(conflicting);
```

### interesting.js
```js
export function isInterestingTag(key) {
    return key !== 'attribution' &&
        key !== 'created_by' &&
        key !== 'source' &&
        key !== 'odbl' &&
        key.indexOf('tiger:') !== 0;
}
```

### cjs output.js
```js
'use strict';

var toCheck = [{highway: 'primary'}, {highway: 'secondary'}];
var tags = {};
var conflicting = false;

toCheck.forEach(function(src) {
	for (var k in src) {
		if (!(k in tags)) {
				tags[k] = src[k];
		}
	}
});

console.log(conflicting);
```

</details>

**Comment by bhousel** [comment]
As a workaround, I found that I could change:

```js
  } else if (tags[k] && isInterestingTag(k) && tags[k] !== src[k]) {
```
to
```js
  } else if (isInterestingTag(k) && tags[k] && tags[k] !== src[k]) {
```

**Comment by zenmumbler** [comment]
I did a quick check on the repl with the rollup versions and this issue seems to have been introduced in 0.61.1, works fine on 0.61.0 and earlier, broken afterwards. For now you may want to downgrade to to either 0.60.7 or 0.61.0. (Sorry, not source knowledgeable enough to fix this)

**Comment by lukastaegert** [comment]
Sorry about the issues, this should be resolved with #2303.

**Comment by lukastaegert** [comment]
Fix released via Rollup v0.63.0, thanks for being patient!

**Comment by bhousel** [comment]
> Fix released via Rollup v0.63.0, thanks for being patient!

No worries, thank you for the fix!  🙇
